impl/src/planning.rs

```diff
@@ -355,20 +364,72 @@ impl<'fetcher> BuildPlanner for BuildPlannerImpl<'fetcher> {
   fn plan_build(
     &mut self,
     settings: &RazeSettings,
+    path_prefix: &PathBuf,
     files: CargoWorkspaceFiles,
     platform_details: Option<PlatformDetails>,
   ) -> Result<PlannedBuild> {
-    let metadata = self.metadata_fetcher.fetch_metadata(&files)?;
+    let metadata = self.metadata_fetcher.fetch_metadata(&files, true)?;
+
+    // Create one combined metadata object which includes all dependencies and binaries
     let crate_catalog = CrateCatalog::new(&metadata)?;
 
-    // Generate additional PlatformDetails
+    // Additionally, fetch metadata for the list of binaries present in raze settings. This
+    // is only supported in Remote mode as it's expected that `vendor` has provided all the sources.
+    let mut bin_crate_catalogs = Vec::new();
+    let mut bin_crate_files: HashMap<String, CargoWorkspaceFiles> = HashMap::new();
+
+    if settings.genmode == GenMode::Remote {
+      let mut bin_metadatas: Vec<Metadata> = Vec::new();
+
+      // Determine the lockfiles dir so we can either reuse existing lockfiles or output new ones
+      let lockfiles_dir = path_prefix.clone().as_path().join("lockfiles");
+
+      // Get a path to the temporary directory where we'll effectively vendor binary dependencies
+      let tempdir = match self.binary_deps_tempdir.as_ref() {
+        Ok(dir) => dir.as_ref(),
+        Err(err) => { 
+          return Err(anyhow!(err.to_string()))
+        },
+      };
+
+      for (package, info) in (&settings.binaries).into_iter() {
+        // Install the package into a temp dir so we can run get it's metadata
+
+        let (_src_dir, bin_workspace_files) = fetch_crate_src(
+          &settings.registry,
+          &PathBuf::from(tempdir),
+          &lockfiles_dir,
+          &package,
+          &info.req().to_string(),
+        )
+        .unwrap();
+
+        bin_metadatas.push(
+          self
+            .metadata_fetcher
+            .fetch_metadata(&bin_workspace_files, false)
```

Comment:
This is the only location where `fetch_metadata` avoids using a temporary directory, and it's not totally clear to me *why* it needs this divergent behavior. What's the rationale / what would be different if it used a temporary directory here too?

Reply:
`fetch_metadata` fails when I pass the newly fetched source of the binary dependencies. There's no reason to use the `tempfile` logic on these directories because they're already in a temporary directory and there's no way they can poison the repository. Because `fetch_metadata` doesn't handle workspaces well (at all?) it's better to just avoid doing this entirely and running `fetch_metadata` from wherever the binary dependency was cloned.